src/constructs/core/migrating.test.ts

```diff
@@ -0,0 +1,105 @@
+import "@aws-cdk/assert/jest";
+import { SynthUtils } from "@aws-cdk/assert";
+import type { BucketProps } from "@aws-cdk/aws-s3";
+import { Bucket } from "@aws-cdk/aws-s3";
+import type { SynthedStack } from "../../../test/utils";
+import { simpleGuStackForTesting } from "../../../test/utils";
+// import type { GuStatefulConstruct } from "./migrating";
+import { GuMigratingResource } from "./migrating";
+import type { GuStack } from "./stack";
+
+describe("GuMigratingResource", () => {
+  /*
+  NOTE: In reality, we'd never directly call `GuMigratingResource.setLogicalId` as the GuConstruct will do that.
+  We're calling it here to test the function in isolation.
+   */
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function -- we are testing `console.warn` being called, we don't need to see the message
+  const warn = jest.spyOn(console, "warn").mockImplementation(() => {});
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function -- we are testing `console.error` being called, we don't need to see the message
+  const error = jest.spyOn(console, "error").mockImplementation(() => {});
+
+  afterEach(() => {
+    warn.mockReset();
+    error.mockReset();
+  });
+
+  test("creating a new resource in a new stack", () => {
```

Comment:
Being annoying here, but could we capture the expected behaviour for when we create a new resource in a new stack? eg

```suggestion
  test("creating a new resource in a new stack produces a new auto-generated ID", () => {
```